index.js

```diff
@@ -66,7 +70,8 @@ module.exports = class Hyperswarm extends EventEmitter {
     this._serverConnections = 0
     this._firewall = firewall
 
-    this.dht.on('network-change', this._handleNetworkChange.bind(this))
+    this._handleNetworkChange = this._handleNetworkChange.bind(this)
+    this.dht.on('network-change', this._handleNetworkChange)
```

Comment:
better to keep a set of sessions on the root, as event emitted are not really meant for N listeners (you'll get leak warnings etc)

Reply:
Set of sessions is a good idea which is now implemented! Leak warning could still happen if `opts.dht` is overused